### PR TITLE
store: implement MVCCDebugger Interface for MVCCLevelDB

### PR DESCRIPTION
**Pull request by Deardrops**
### What problem does this PR solve? 

For improving test coverage of `server` module, we should add missing functions for `txn debugger` apis, and then we could cover these apis in unit test cases.
```golang
router.Handle("/mvcc/key/{db}/{table}/{handle}", mvccTxnHandler{tikvHandlerTool, opMvccGetByKey})
router.Handle("/mvcc/txn/{startTS}/{db}/{table}", mvccTxnHandler{tikvHandlerTool, opMvccGetByTxn})
router.Handle("/mvcc/hex/{hexKey}", mvccTxnHandler{tikvHandlerTool, opMvccGetByHex})
router.Handle("/mvcc/index/{db}/{table}/{index}/{handle}", mvccTxnHandler{tikvHandlerTool, opMvccGetByIdx})
```

### What is changed and how it works?

Implement `MVCCDebugger` Interface for `MVCCLevelDB` in `store/mocktikv` package
```golang
type MVCCDebugger interface {
	MvccGetByStartTS(startKey, endKey []byte, starTS uint64) (*kvrpcpb.MvccInfo, []byte)
	MvccGetByKey(key []byte) *kvrpcpb.MvccInfo
}
```

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - None

Related changes

 - None

### CODE REVIEW

File: store/mockstore/mocktikv/mvcc_leveldb.go [hunk 1]
```diff
@@ -1273,3 +1273,95 @@ func (mvcc *MVCCLevelDB) doRawDeleteRange(startKey, endKey []byte) error {
 
 	return mvcc.db.Write(batch, nil)
 }
+
+// MvccGetByStartTS implements the MVCCDebugger interface.
+func (mvcc *MVCCLevelDB) MvccGetByStartTS(startKey, endKey []byte, starTS uint64) (*kvrpcpb.MvccInfo, []byte) {
+	mvcc.mu.RLock()
+	defer mvcc.mu.RUnlock()
+
+	pairs := mvcc.ReverseScan(nil, nil, 1, starTS, kvrpcpb.IsolationLevel_RC)
```
Comment (crazycs520) [inline store/mockstore/mocktikv/mvcc_leveldb.go]: s/IsolationLevel_RC /IsolationLevel_SI ?

File: store/mockstore/mocktikv/mvcc_leveldb.go [hunk 2]
```diff
@@ -1273,3 +1273,95 @@ func (mvcc *MVCCLevelDB) doRawDeleteRange(startKey, endKey []byte) error {
...
+	var writes []*kvrpcpb.MvccWrite
+	var values []*kvrpcpb.MvccValue
+	for iter.Valid() {
+		ok, err := dec2.Decode(iter)
+		if err != nil {
+			return nil
+		}
+		if !ok {
+			iter.Next()
+			continue
```
Comment (crazycs520) [inline store/mockstore/mocktikv/mvcc_leveldb.go]: should break?
Otherwise, the iter will scan to end value?

Comment (Deardrops) [reply store/mockstore/mocktikv/mvcc_leveldb.go]: https://github.com/pingcap/tidb/blob/a4d6f8863617f2179d9a616f7e8a9848ad4e8f3e/store/mockstore/mocktikv/mvcc_leveldb.go#L223-L225
No, the `valueDecoder.Decode()` will return false if current iter key not match the excepted key.
So we need traverse the entire iteratror, and skip these unmatched keys.

**Comment by Deardrops** [comment]
@tiancaiamao @bb7133 PTAL, this PR will ployfill missing function in mocktikv, and finaly improve the test coverage of `server` module.

File: store/mockstore/mocktikv/mvcc_leveldb.go (hunk 2 above)
Comment (tiancaiamao) [reply store/mockstore/mocktikv/mvcc_leveldb.go]: > So we need traverse the entire iteratror, and skip these unmatched keys.

I'm confused about that. It actually scans to the last one if you write code like that.

Comment (tiancaiamao) [reply store/mockstore/mocktikv/mvcc_leveldb.go]: We need to scan all the versions of a key, but not all keys.

File: store/mockstore/mocktikv/mvcc_leveldb.go [hunk 3]
```diff
@@ -1273,3 +1273,95 @@ func (mvcc *MVCCLevelDB) doRawDeleteRange(startKey, endKey []byte) error {
 
 	return mvcc.db.Write(batch, nil)
 }
+
+// MvccGetByStartTS implements the MVCCDebugger interface.
+func (mvcc *MVCCLevelDB) MvccGetByStartTS(startKey, endKey []byte, starTS uint64) (*kvrpcpb.MvccInfo, []byte) {
```
Comment (crazycs520) [inline store/mockstore/mocktikv/mvcc_leveldb.go]: `startKey, endKey` is useless?

**Comment by Deardrops** [comment]
@crazycs520 @tiancaiamao Address comment, PTAL

File: store/mockstore/mocktikv/mvcc_leveldb.go (hunk 3 above)
Comment (tiancaiamao) [reply store/mockstore/mocktikv/mvcc_leveldb.go]: It's used in line 1282

**Comment by tiancaiamao** [comment]
LGTM

File: server/http_status.go [hunk 4]
```diff
@@ -103,11 +103,15 @@ func (s *Server) startHTTPServer() {
 		router.Handle("/regions/meta", regionHandler{tikvHandlerTool}).Name("RegionsMeta")
 		router.Handle("/regions/hot", regionHandler{tikvHandlerTool}).Name("RegionHot")
 		router.Handle("/regions/{regionID}", regionHandler{tikvHandlerTool})
+	}
+
+	if s.cfg.Store == "tikv" || s.cfg.Store == "mocktikv" {
 		router.Handle("/mvcc/key/{db}/{table}/{handle}", mvccTxnHandler{tikvHandlerTool, opMvccGetByKey})
 		router.Handle("/mvcc/txn/{startTS}/{db}/{table}", mvccTxnHandler{tikvHandlerTool, opMvccGetByTxn})
```
Comment (crazycs520) [inline server/http_status.go]: There is some problem when I use this HTTP API when store is `mocktikv`.

Comment (Deardrops) [reply server/http_status.go]: Comment Addressed, test passed in local with single TiDB + Mocktikv, PTAL

File: store/mockstore/mocktikv/mvcc_leveldb.go [hunk 5]
```diff
@@ -1273,3 +1273,107 @@ func (mvcc *MVCCLevelDB) doRawDeleteRange(startKey, endKey []byte) error {
...
+		iter.Next()
+	}
+	if key == nil {
+		return nil, nil
+	}
+
+	return mvcc.MvccGetByKey(key), key
+}
+
+var valueTypeOpMap = map[mvccValueType]kvrpcpb.Op{
```
Comment (lonng) [inline store/mockstore/mocktikv/mvcc_leveldb.go]: Prefer: 
```suggestion
var valueTypeOpMap = [...]kvrpcpb.Op{
```

**Pull request merged**